### ISSUE TITLE
Mutable reference on owner in builder

### ISSUE THREAD

**Issue by gwenn** [Voultapher/self_cell#59]
Would it be possible to have `&'a mut $Owner`:
```rust
fn try_new<Err>(
    owner: $Owner,
    dependent_builder: impl for<'a> ::core::ops::FnOnce(&'a mut $Owner) -> Result<$Dependent<'a>, Err>
) -> Result<Self, Err>
```
instead of
```rust
fn try_new<Err>(
    owner: $Owner,
    dependent_builder: impl for<'a> ::core::ops::FnOnce(&'a $Owner) -> Result<$Dependent<'a>, Err>
) -> Result<Self, Err>
```

(use case: https://github.com/rusqlite/rusqlite/pull/1462/files#diff-c595ea0bc5c6bee5e8dfc649ad92279225e2e9a354b8f4d355fbaf4012ebe1e8R19)

**Comment by Voultapher** [comment]
Hi, this has been discussed here a couple of times, for example here https://github.com/Voultapher/self_cell/pull/36. It's one of the main differences to ouroboros API wise, allowing it easily leads to [UB](https://github.com/Voultapher/self_cell/pull/36#issuecomment-1336171949), the available API would need to be more variable than it is now. Although now that I'm revisiting the topic I have a new idea how it could be tackled without making the API variable.

**Comment by Voultapher** [comment]
@gwenn could you please the main branch and let me know if the added `MutBorrow` works for your use-cases. This example should explain how to use it.

**Comment by gwenn** [comment]
Tested successfully:
https://github.com/rusqlite/rusqlite/pull/1462/commits/93787c68551f8526cd5cec71d0a9967814f94e7e
Thanks.

**Comment by Voultapher** [comment]
Cool!

I plan on doing another feature, after which I plan on releasing a new version that contains `MutBorrow`. Will close this issue once the new version is released.

**Comment by hniksic** [comment]
Thanks for implementing this feature - please consider uploading a release that contains it. It'd be nice to replace uses of ouroboros with self-cell, as self-cell generates much less code. But the switch is blocked by cases where the builder requires a mutable owner (and it doesn't make sense to use two self-referential crates in the same workspace).

**Comment by Voultapher** [comment]
Thanks for the reminder, I have to admit to myself that the idea will just not be ready soon so while it would have been nice to have a bigger 1.1 release, let's just release this.

**Comment by Voultapher** [comment]
Done https://github.com/Voultapher/self_cell/releases/tag/v1.1.0.

**Comment by gwenn** [comment]
@Voultapher 
Wonderful !
